File: 2023/domaci/41-textovy-dekomprimator/volfmatous/main.cs
using System;

internal class Program
{
    static void Main(string[] args)
    {
        string text = Console.ReadLine();
        bool compressing = Console.ReadLine() == "compress";

        if (compressing)
        {
            int currentLetterCount = 1;

            for (int i = 0; i < text.Length; i++)
            {
                if (i < text.Length - 1 && text[i] == text[i + 1])
                {
                    currentLetterCount++;
                }
                else
                {
                    Console.Write(text[i]);
                    if (currentLetterCount > 1)
                    {
                        Console.Write(currentLetterCount);
                    }

                    currentLetterCount = 1;
                }
            }
        }
        else
        {
            for (int i = 0; i < text.Length; i++)
            {
                Console.Write(text[i]);

                string countBuffer = string.Empty;
                for (int j = i; j < text.Length; j++)
                {
                    if (j < text.Length - 1 && char.IsDigit(text[j + 1]))
                    {
                        countBuffer += text[j + 1];
                    }
                    else
                    {
                        if (countBuffer.Length > 0)
                        {
                            for (int k = 0; k < int.Parse(countBuffer) - 1; k++)
                            {
                                Console.Write(text[i]);
                            }
                        }

                        i = j;
                        break;
                    }
                }
            }
        }

        Console.WriteLine();
    }
}

File: 2023/domaci/41-textovy-dekomprimator/volfmatous/main.js
const readline = require("readline");
const rl = readline.createInterface({
    input: process.stdin,
    output: process.stdout,
    terminal: false,
});

let linesRead = 0;
const lines = [];

async function readLine() {
    if (lines.length < 1) {
        for await (const line of rl) {
            lines.push(line);
        }
    }

    linesRead++;
    return lines[linesRead - 1];
}

main().then();

async function main() {
    const text = await readLine();
    const compressing = await readLine() === "compress";

    if (compressing) {
        let currentLetterCount = 1;

        for (let i = 0; i < text.length; i++) {
            if (i < text.length - 1 && text[i] === text[i + 1]) {
                currentLetterCount++;
            } else {
                process.stdout.write(text[i]);
                if (currentLetterCount > 1) {
                    process.stdout.write(currentLetterCount.toString());
                }

                currentLetterCount = 1;
            }
        }
    } else {
        for (let i = 0; i < text.length; i++) {
            process.stdout.write(text[i]);

            let countBuffer = "";
            for (let j = i; j < text.length; j++) {
                if (j < text.length - 1 && text[j + 1].charCodeAt(0) >= 48 && text[j + 1].charCodeAt(0) <= 57) {
                    countBuffer += text[j + 1];
                } else {
                    if (countBuffer.length > 0) {
                        const count = parseInt(countBuffer, 10);
                        for (let k = 0; k < count - 1; k++) {
                            process.stdout.write(text[i]);
                        }
                    }

                    i = j;
                    break;
                }
            }
        }
    }

    process.stdout.write("\n");
}
